src/Engine/Engine.cpp

```diff
@@ -538,33 +514,43 @@ void Engine::LogEngineBuildInfo() {
 }
 
 //----- (0044EA5E) --------------------------------------------------------
-void Engine::PickMouse(float fPickDepth, unsigned int uMouseX,
-                       unsigned int uMouseY, bool bOutline,
-                       Vis_SelectionFilter *sprite_filter,
-                       Vis_SelectionFilter *face_filter) {
+Vis_PIDAndDepth Engine::PickMouse(float fPickDepth, unsigned int uMouseX, unsigned int uMouseY,
+                                  Vis_SelectionFilter *sprite_filter, Vis_SelectionFilter *face_filter) {
     if (uMouseX >= (signed int)pViewport->uScreen_TL_X &&
         uMouseX <= (signed int)pViewport->uScreen_BR_X &&
         uMouseY >= (signed int)pViewport->uScreen_TL_Y &&
         uMouseY <= (signed int)pViewport->uScreen_BR_Y) {
-        vis->PickMouse(fPickDepth, uMouseX, uMouseY, sprite_filter, face_filter);
-
-        if (bOutline)
-            OutlineSelection();
+        return vis->PickMouse(fPickDepth, uMouseX, uMouseY, sprite_filter, face_filter);
+    } else {
+        return Vis_PIDAndDepth();
     }
 }
 
 //----- (0044EB12) --------------------------------------------------------
-bool Engine::PickKeyboard(float pick_depth, bool bOutline, Vis_SelectionFilter *sprite_filter,
-                          Vis_SelectionFilter *face_filter) {
+Vis_PIDAndDepth Engine::PickKeyboard(float pick_depth, Vis_SelectionFilter *sprite_filter, Vis_SelectionFilter *face_filter) {
     if (current_screen_type == SCREEN_GAME) {
-        bool r = vis->PickKeyboard(pick_depth, &vis->default_list, sprite_filter, face_filter);
-
-        if (bOutline)
-            OutlineSelection();
-        return r;
+        return vis->PickKeyboard(pick_depth, sprite_filter, face_filter);
+    } else {
+        return Vis_PIDAndDepth();
     }
-    return false;
 }
+
+Vis_PIDAndDepth Engine::PickMouseInfoPopup() {
+    Pointi pt = mouse->GetCursorPos();
+    // TODO(captainurist): having a different depth here makes no sense.
```

Comment:
Make sense to me that you would be able to Id targets at a longer range than you could shoot them from.

Reply:
Yeah I'll reformulate the TODO there